fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -361,8 +368,18 @@ public int compareLiteral(LiteralExpr expr) {
         if (expr == MaxLiteral.MAX_VALUE) {
             return -1;
         }
+
+        long thisExprLongValue = getLongValue();
+        long otherExprLongValue = expr.getLongValue();
+        if (!getType().equals(expr.getType())) {
+            if (getType().equals(Type.DATE)) {
```

Comment:
Are you sure the `expr` is either `DATE` or `DATETIME` type?
May it be a Long value? for example.

Reply:
Thank you for reminding.
`Expr` could be other type besides `DATE` and `DATETIME`. I find the constructor or `init` method separately carry out `DATE` type. Other types are treated as `DATETIME` indeed, even though the assigned type is not `DATETIME`.

Reply:
Thank you for reminding.
`Expr` could be other type besides `DATE` and `DATETIME`. I find the constructor or `init` method separately carry out `DATE` type. Other types are treated as `DATETIME` indeed, even though the assigned type is not `DATETIME`.

Reply:
Could you please give me some advice on how to process other types reasonably?
What I can think of now is to process `DATE` type separately, because the constructor or init-method or getLongValue-method only process `DATE` type separately too. And other types (e.g. LONG), the assigned type is not `DATEIME`, but in fact they are still processed as `DATETIME`.